samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart

```diff
@@ -427,9 +427,10 @@ class PetApi {
             'application/x-www-form-urlencoded',
         ];
 
-        final formData = <String, dynamic>{};
-        formData['name'] = parameterToString(_serializers, name);
-        formData['status'] = parameterToString(_serializers, status);
+        final formData = <String, dynamic>{
+            if (name != null) r'name': parameterToString(_serializers, name),
```

Comment:
I would put null-check inside `parameterToString` to remove repeating code, what's the effect of `formData['name'] = null` ?

Reply:
There is one but it defaults to `''`. I am unsure what the consequences are if I remove this.

Reply:
Also the null-check is only done when the param is not `required`.